### ISSUE TITLE
Server browser connects as soon a server is selected.

### ISSUE THREAD

**Issue by ZehM4tt** [OpenRCT2/OpenRCT2#6091]
**Version:** 0.1.1
**Commit/Build:** 1fed4f0

As soon one selects a server in the list it automatically connects while a context menu shows up, however there is no time to select/click anything as its already establishing a connection at the specific moment.

- [ ] Reproducible in RCT2 (vanilla)?
- [x] Multiplayer?

**Steps to reproduce:**
1. Open the server browser
2. Select any server

**Comment by janisozaur** [comment]
click-and-hold

**Comment by PFCKrutonium** [comment]
Eh, this catches me often enough as well that I feel the need to support this change.

**Comment by janisozaur** [comment]
In my opinion it should stay as is. Given that you already browse for servers, it is clear the intention is to join one – the only other option from the context menu is to add to favourites, which would by far be used many times fewer than the join one.

If we were to switch from what it is doing currently, this would interrupt current flow of joining the server. Even in the case user wanted to add it to the favourites, but joined by mistake they can:
1. cancel connection when it is still loading,
2. disconnect from the server and then add it later on.
That said, I can see a scenario where such a user would just stay on the server he already intended to add to favourites, play for some time and just add it next time ha launches the game.

**Comment by Gymnasiast** [comment]
There is also the third option of adding a favourite button. There current click-and-hold menu on a list item is a bit of an oddball w/r/t UI design. 

But I agree with janisozaur it is not a good idea to change the menu to stay open, as that will indeed interrupt the flow of 99% of use cases to better support the other 1%.

**Comment by ZehM4tt** [comment]
So why can't we do it the normal way, double left click to connect and right click for context.

**Comment by Gymnasiast** [comment]
> So why can't we do it the normal way, double left click to connect and right click for context. 

We could do that. I think a separate is slightly clearer, but there isn't anything big holding us back AFAIK.

> Also try to open and hold this menu on low FPS like 10-5, this is a nightmare of design.

Something to fix, then. We can't remove all self-closing dropdowns just because they don't work properly in low FPS conditions.

**Comment by ZehM4tt** [comment]
I'm just talking about the server browser nothing else.

**Comment by Gymnasiast** [comment]
No, but your point about holding menus on low FPS either applies to all self-closing dropdowns or doesn't apply at all.

**Comment by zsilencer** [comment]
:smile: We have a issue for literally the exact opposite of this #2760

**Comment by 733737** [comment]
Are you going to add a cancel button to the connecting dialog box?
